website_sale_wishlist/models/wishlist_product.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from openerp import api, fields, models
+from openerp.http import request, root
+
+
+class WishlistProduct(models.Model):
+    _name = "wishlist.product"
+    _sql_constrains = [
+        ("session_or_user_id",
+         "CHECK(session IS NULL != user_id IS NULL)",
+         "Need a session or user for wishlisting a product, but never both."),
+        ("product_unique_session",
+         "UNIQUE(product_tmpl_id, session)",
+         "Duplicated wishlisted product for this session."),
+        ("product_unique_user_id",
+         "UNIQUE(product_tmpl_id, user_id)",
+         "Duplicated wishlisted product for this user."),
+    ]
+
+    product_tmpl_id = fields.Many2one(
+        comodel_name="product.template",
```

Comment:
Odoo works at template level in the e-commerce (product list, product detail). Variants are reflected through selection options in the template detail page.